connectors/sources/sharepoint_online.py

```diff
@@ -879,258 +1003,240 @@ async def validate_config(self):
 
-        site_web_url = site["webUrl"]
+        Returns:
+            list: access control list for a given site
+            [
+                "user:spo-user",
+                "email:some.user@spo.com",
+                "group:1234-abcd-id"
+            ]
+        """
 
-        sharepoint_groups = []
+        if not self._dls_enabled():
+            return site
 
-        async for sharepoint_group in self.client.site_groups(site_web_url):
-            sharepoint_groups.append(sharepoint_group)
+        access_control = set()
 
-        sharepoint_groups = list(
-            map(
-                lambda group: _prefix_group(group),
-                filter(
-                    lambda group_title: group_title is not None,
-                    map(
-                        lambda group: group.get("Title"),
-                        sharepoint_groups,
-                    ),
-                ),
-            )
-        )
+        async for user_information in self.client.user_information_list(site["id"]):
+            user = user_information["fields"]
 
-        site_users = []
-
-        async for site_user in self.client.site_users(site_web_url):
-            site_users.append(site_user)
-
-        site_users = list(
-            map(
-                lambda user: _prefix_user(user),
-                filter(
-                    lambda user_name: user_name is not None,
-                    map(
-                        lambda user: user.get("UserPrincipalName"),
-                        site_users,
-                    ),
-                ),
-            )
-        )
+            if is_domain_group(user):
+                domain_group_id = _domain_group_id(user["Name"])
 
-        return self._decorate_with_access_control(site, sharepoint_groups + site_users)
+                if domain_group_id:
+                    group = await self.client.group(domain_group_id)
 
-    async def _with_drive_item_access_control(self, site_drive, drive_item):
-        """
-        Sample document returned:
-        {
-            "grantedToV2": {
-                "siteGroup": {
-                    "loginName": "some_group"
-                    }
-                },
-            "grantedTo": {
-                "siteGroup": {
-                    "loginName": "user"
-                    }
-                },
-                "user": {
-                    "email": USER_2
-                }
-            }
+                    if len(group) > 0:
+                        access_control.add(_prefix_group(group["id"]))
 
-        We're extracting "loginName" or "email" for groups or users inside "grantedToV2" or "grantedTo".
-        """
+                    async for member_email_or_username in _emails_and_usernames_of_domain_group(
+                        domain_group_id, self.client.group_members
+                    ):
+                        access_control.add(member_email_or_username)
```

Comment:
After talking, instead of holding these in an in-memory cache - just remove this logic for getting the users from groups. We'll add the explicit groups to `_allow_access_control`, and the explicit users to `_allow_access_control`, but not the implicit users.